base_layer/core/src/block.rs

```diff
@@ -238,16 +238,11 @@ impl BlockBuilder {
     /// This functions add the provided transactions to the block
     pub fn with_transactions(mut self, txs: Vec<Transaction>) -> Self {
         let mut iter = txs.into_iter();
-        loop {
-            match iter.next() {
-                Some(tx) => {
-                    self = self.add_inputs(tx.body.inputs);
-                    self = self.add_outputs(tx.body.outputs);
-                    self = self.add_kernels(tx.body.kernels);
-                    self.header.total_kernel_offset = self.header.total_kernel_offset + tx.offset;
-                },
-                None => break,
-            }
+        while let Some(tx) = iter.next() {
```

Comment:
Any reason this can't just be a `for` loop?

Reply:
Thanks. Rewritten as a for loop.